components/andes/org.wso2.carbon.andes.core/src/main/java/org/wso2/carbon/andes/core/QueueManagerServiceImpl.java

```
@@ -537,16 +598,89 @@ private Queue getQueue(String nameOfQueue, String userName, String accessKey) th
+     *
+     * @param queueName queue name
+     * @param queueId   Id given to the queue
+     * @param userRealm User's Realm
+     * @throws QueueManagerException
+     */
+    private static void authorizePermissionsToLoggedInUser(String queueName,
+                                                           String queueId,
+                                                           UserRealm userRealm) throws
+                                                                                QueueManagerException {
+        //For registry we use a modified queue name
+        String newQueueName = queueName.replace("@", AT_REPLACE_CHAR);
+
+        String username = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        try {
+            String roleName = UserCoreUtil.addInternalDomainName(QUEUE_ROLE_PREFIX +
+                                                                 queueName.replace("/", "-"));
+            UserStoreManager userStoreManager = CarbonContext.getThreadLocalCarbonContext()
+                    .getUserRealm().getUserStoreManager();
+
+            if (!userStoreManager.isExistingRole(roleName)) {
+                String[] user = {MultitenantUtils.getTenantAwareUsername(username)};
+
+                userStoreManager.addRole(roleName, user, null);
+                userRealm.getAuthorizationManager().authorizeRole(roleName, queueId,
+                                                                  PERMISSION_CHANGE_PERMISSION);
+                userRealm.getAuthorizationManager().authorizeRole(
+                        roleName, queueId, TreeNode.Permission.CONSUME.toString().toLowerCase());
+                userRealm.getAuthorizationManager().authorizeRole(
+                        roleName, queueId, TreeNode.Permission.PUBLISH.toString().toLowerCase());
+            } else {
+                log.warn("Unable to provide permissions to the user, " +
```

Should we throw an exception here?